ocaml/parsing/parser_types.mli

```diff
@@ -0,0 +1,40 @@
+(* Tracks types that the parser uses that get exposed when compiling via Menhir. This
+   prevents the compiled parser.mli from having undeclared types, since it doesn't
+   copy the implementation section from parser.mly, causing types declared in
+   parser.mly and then exposed by Menhir in function signatures in parser.mli
+   will be unbound. *)
```

Comment:
This comment is hard to make sense of. I think it should just say that these are types used by the parser, and that they need to be placed in a separate module because when you use `--inspection` the parser's interface refers to the result type of every parsing rule.